### ISSUE TITLE
[Fleeing for Assist]: NPCs on Low HP need 1.666% reduced speed per 1% health below 30% HP [$30]

### ISSUE THREAD

**Issue by Stifler82** [azerothcore/azerothcore-wotlk#5342]
**EDITED**: to add the outcome of the discussion here, provided by @T1ti 

So TLDR how fleeing works on classic : 

- Fleeing doesn't change the creature's speed at all, it should still have the same speed as if it was not fleeing

- **But currently AC is missing the low health speed reduction mechanic, which is about -2.5% reduced speed per 1% hp below 20%.** 
This is probably the msot important point

- If you FEAR a creature which doesn't have reduced move speed(above 20% hp) like with warlock's fear, the creature will run at full speed. 

- **as a recap, fleeing state never affects the creature's speed, creature's speed is only affected by external factors like the low health speed reduction mechanic, or other slwos applied by players**

- Creatures seem to run to other creatures while fleeing if they get close to them, but the original fleeing direction seems completely random(my personal analysis). When they do this, they stop at the creature and trigger a call for help, after a small delay the creatures will aggro and the fleeing creature will stop fleeing exactly at the same time the new creatures aggro.

- basically, fleeing movement is compeltely RNG, UNTIL they get close to another creature, then they should? run to it and do a call for help.(previous point)

-----------
Originally reported https://github.com/chromiecraft/chromiecraft/issues/183

##### ISSUE: FACTION SIDE:
BOTH

##### CONTENT PHASE:
ALL

##### SMALL DESCRIPTION:
Half the times, when you're fighting a humanoid mob that will flee at low health, once they reach the threshold, they will IMMEDIATELY take off at a very high speed, even faster than regular running, towards an enemy pack. Once they get anywhere near another mob, they will INSTANTLY pull the mob.

This is blatantly wrong, even by Classic standards.

This is really frustrating and punishing to deal with. Sure, you can say "just save your burst for when they get really low", but that's not a luxury that can always be afforded, especially by certain classes. And when that happens: death.

##### EXPECTED BLIZZLIKE BEHAVIOUR:
https://www.youtube.com/watch?v=CuvYOo1jG1k

I don't think ALL the mobs are supposed to run at ludicrous speeds. Even for the ones that do, they seem way too fast.

What's 100% wrong is that the mobs pull way too quickly once they get to other mobs. If you're ranged, the fleeing mob can get into kissing range with the non-pulled mob, and you can still kill the fleeing mob without the other pulling. Here, not quite...

##### CURRENT BEHAVIOUR:
See "Small Description".

##### STEPS TO REPRODUCE THE PROBLEM:
Watch the video I linked above, then recreate it using a Paladin at-level with leveling gear. Notice how you get destroyed by the mobs in the cave.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/6a7155eb7e82309ac34a38eec65eb7663b0605a0

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by FrancescoBorzi** [comment]
isn't this 1-19? Makrura Clacker is in Durotar:

https://classic.wowhead.com/npc=3103/makrura-clacker

**Comment by wizzymore** [comment]
Hi,

This behaviour is completely normal.
It was even discussed on WoW Classic a while ago before launch and Blizzard released this in the "It's a feature not a bug" list [here](https://us.forums.blizzard.com/en/wow/t/wow-classic-%E2%80%9Cnot-a-bug%E2%80%9D-list-updated-dec-3-2020/175887)
They are supposed to run faster than you so you can't catch them, and, this is even realistic as in real life you would run faster because of the adrenaline rush.

Also, what you are showing there is from retail, you can not compare retail with this in our current example.

I'll say it a blizzard said it, this is a feature, not a bug.

**Comment by Stifler82** [comment]
> Hi,
> 
> This behaviour is completely normal.
> It was even discussed on WoW Classic a while ago before launch and Blizzard released this in the "It's a feature not a bug" list [here](https://us.forums.blizzard.com/en/wow/t/wow-classic-%E2%80%9Cnot-a-bug%E2%80%9D-list-updated-dec-3-2020/175887)
> They are supposed to run faster than you so you can't catch them, and, this is even realistic as in real life you would run faster because of the adrenaline rush.
> 
> Also, what you are showing there is from retail, you can not compare retail with this in our current example.
> 
> I'll say it as blizzard said it, this is a feature, not a bug.

Thank you so much for your feedback!

**Comment by apav** [comment]
> Hi,
> 
> This behaviour is completely normal.
> It was even discussed on WoW Classic a while ago before launch and Blizzard released this in the "It's a feature not a bug" list [here](https://us.forums.blizzard.com/en/wow/t/wow-classic-%E2%80%9Cnot-a-bug%E2%80%9D-list-updated-dec-3-2020/175887)
> They are supposed to run faster than you so you can't catch them, and, this is even realistic as in real life you would run faster because of the adrenaline rush.
> 
> Also, what you are showing there is from retail, you can not compare retail with this in our current example.
> 
> I'll say it as blizzard said it, this is a feature, not a bug.

Are you sure by "feared" they don't mean spells and abilities that fear? I can only recall low health enemies hobbling away in WotLK. If they did actually run at full speed or even greater, I think I would remember dying many times and confined spaces like caves being this difficult because due to me not being able to kill them in time.

**Comment by eTzmNcbkrng** [comment]
Agree with @apav here, feared mobs/players have always run faster than normal (unless under the effects of movement slowing).
I think there is a difference between feared and fleeing.  
That also goes directly against the video source I provided from live servers at the time.

Here's a response to that comment showing they're also interpreting it as a Warlock's Fear:
![image](https://user-images.githubusercontent.com/26512802/115160262-7e339300-a0eb-11eb-9d85-bf3d54a23252.png)

**Comment by Stifler82** [comment]
> Agree with @apav here, feared mobs/players have always run faster than normal (unless under the effects of movement slowing).
> I think there is a difference between feared and fleeing.
> That also goes directly against the video source I provided from live servers at the time.
> Edit: Noticed the video I mentioned is on ChromieCraft Issue, not here - https://www.youtube.com/watch?v=8dNMBWh2wxY - 6:55 and 8:10.
> 
> Here's a response to that comment showing they're also interpreting it as a Warlock's Fear:
> ![image](https://user-images.githubusercontent.com/26512802/115160262-7e339300-a0eb-11eb-9d85-bf3d54a23252.png)

Idk, could be or might be that they say "Fear" as concept "In General" of Fear, so they are not referring to a specific mechanic, they are simply explaining that "Fear" is fear, and when you have "Fear", you run away.
So they found it "superfluous" to specify..

But as i said: idk, so i'm not sure 100%.

**Comment by Stifler82** [comment]
>I think there is a difference between feared and fleeing.

Among the definitions, I can't see **"Walk"**

![image](https://user-images.githubusercontent.com/1884642/115160713-a4c4e000-a091-11eb-8dcf-d053bb817ba0.png)

**Comment by T1ti** [comment]
I personally tought it was also an issue a few weeks ago so I had tested it
it looks fine on AC, I found that on Classic some mobs just run very fast like murlocs, some run slow
I found the same patterns in AC

**Comment by 55Honey** [comment]
How can a Classic reference outclass a 3.3.5 retail Video? I am Not convinced.

**Comment by Branel** [comment]
You literally used a **frost mage** as a source to prove that they should not run, did you ever cast frostbolt and see what it does - is this selection bias or ignorance?

Now you shouldn't argue about definitions and word usage because Blizzard is known to use them vaguely, tooltips are constantly proven to be wrong and you shouldn't believe it just because that is what it says in the game.

@apav is right within a reasonable doubt because this behavior is recongnized by anyone that played back in 2010. The NPC would run depending on the amount of health they have left, so if they have 20% they would run really fast, and if they have 1% they would have walking speed. This makes sense right? You don't see people with critical wounds running marathons.

[A recent discussion about this](https://www.reddit.com/r/wowservers/comments/52p5pe/humanoid_fleeing_behaviour/)
[Some people that back my statement](https://www.reddit.com/r/rebirthwow/comments/2gkx63/fleeing_mobs/)

**Comment by Stifler82** [comment]
> You literally used a **frost mage** as a source to prove that they should not run, did you ever cast frostbolt and see what it does - is this selection bias or ignorance?
> 
> Now you shouldn't argue about definitions and word usage because Blizzard is known to use them vaguely, tooltips are constantly proven to be wrong and you shouldn't believe it just because that is what it says in the game.
> 
> @apav is right within a reasonable doubt because this behavior is recongnized by anyone that played back in 2010. The NPC would run depending on the amount of health they have left, so if they have 20% they would run really fast, and if they have 1% they would have walking speed. This makes sense right? You don't see people with critical wounds running marathons.
> 
> [A discussion about this](https://www.reddit.com/r/wowservers/comments/52p5pe/humanoid_fleeing_behaviour/)
> [Some comments that back my statement](https://www.reddit.com/r/rebirthwow/comments/2gkx63/fleeing_mobs/)


@Branel
Thank you so much for your comment, I'm glad to see that there are still people who can make the brain work.
What you just explained makes **PERFECT sense.**
keyword: **"PERFECT"**
and i agree with you **101%**

**[Disclaimer]:**
`This content refers to a generalized personal thought and does not refer to any participant of this conversation`
Unfortunately, I realize that it is not easy for every person in the world to do this, because cognitive processes need a neuronal interaction with a minimum of 2 neurons participating in the interaction, unfortunately when you only have one it is like pulling a ball against the wall and playing alone there is not much interaction..

**Comment by 55Honey** [comment]
>  - is this selection bias or ignorance?
> [Some comments that back my statement](https://www.reddit.com/r/rebirthwow/comments/2gkx63/fleeing_mobs/)

First off, this is not the kind of language, that i expect from a discussion on this repo. If you think this is the way to go, i am pretty sure you're advised to read about our [Code of conduct.](https://github.com/azerothcore/azerothcore-wotlk/discussions/4022) @Branel 

On topic, in the videos provided it is visible that the mobs use a walking animation while fleeing, not the run animation. This is not related to the frostbolt as far as i can tell. I'll be happily proven wrong. Just not by a classic source unrelated to that fact.

If the sources regarding mob health being related to run speed are right, there is still an issue, beacause as of now, they run at full speed even at 1%.

**Comment by Stifler82** [comment]
> If the sources regarding mob health being related to run speed are right, there is still an issue, beacause as of now, they run at full speed even at 1%.

I agree, in this..

**Comment by wizzymore** [comment]
Guys, such a big topic regarding this minor stuff.

The logic of running away in wotlk.
Less health slow movement speed
More health bigger movement speed
Movement speed boost affected by the base speed of npc as well, that's why murlocs are Usain Bolt. ( i am not totally sure that this logic is implementez 100% in our core, but yes, this is a correct behaviour )

Also, the mage in the video has frost armor.

About clasic and retail 3.3.5, you can compare vanill with tbc and 3.3.5 in 90% of cases in the old world. The major changes got done in Cataclysm.

The current behaviour is correct.
Also, fear means both humanoids getting feared also the spell fear.

In conclusion, humanoids RUN away when low health, and this RUN is not the normal base speed that they come to you, is bigger.

**Comment by 55Honey** [comment]
> If the sources regarding mob health being related to run speed are right, there is still an issue, beacause as of now, they run at full speed even at 1%.

Can somebody please test if the behaviour for a handful of different, fleeing mobs is the same for 1% and 20% health?

Reading the comments, i assume we can either safely close this or edit the initial issue based on these results.

**Comment by Stifler82** [comment]
The HP percentage does not affect the speed in Flee

https://user-images.githubusercontent.com/1884642/115545828-2a1be080-a29c-11eb-866d-16a615b81ed1.mp4

**Comment by FrancescoBorzi** [comment]
Regardless of what my opinion is about the technical issue here, this project is about software development and there are [some rules](https://github.com/azerothcore/azerothcore-wotlk/discussions/4022) that everyone should always respect.

I don't want to see people talking about other people being ignorant, about using or not the brain, about how neurons work, etc... This is a space to discuss bugs, fixes and technical topics, always respecting other users even when they technically wrong about something.

General note for everyone: If you are not able to be polite with other users, then this is not a project for you.

**Comment by T1ti** [comment]
FYI, mobs walking at reduced speed doesn't only apply in fear/fleeing status
Mobs also run slower in combat if they are low healh, it's a global mechanic and not related to fleeing at all.

The threshold at which they are moving slower is about 20%
They slowly run slower based on how much hp they lost below 20%

I noticed that at 1% health it's rougly 50% speed
So they lose about 2.5% speed per 1% hp below 20%

**Comment by T1ti** [comment]
source : me using the print movespeed script on classic
![image](https://user-images.githubusercontent.com/40864460/115591061-feb3e880-a2d1-11eb-9878-c6e781fcfd49.png)
![image](https://user-images.githubusercontent.com/40864460/115591072-04a9c980-a2d2-11eb-8251-0dffc5d2f239.png)
![image](https://user-images.githubusercontent.com/40864460/115591090-083d5080-a2d2-11eb-8186-1992900cec39.png)


This creature starts at about 8.0 speed, it reaches about 4.7 when I put him very low, i think that theorycaly 0% health would be about 50% speed as I got some extremly low health mobs to about 45% reduced speed

**Comment by T1ti** [comment]
It works exactly the same in fleeing, I didn't notice any speed diffference that their intended speed when they were fleeing

**Comment by FrancescoBorzi** [comment]
what's the status of this?

**Comment by Maxproz** [comment]
Could try doing a *= on the velocity based on the creatures current health in MoveSplineInit.cpp. 
I think the fleeing movement generator calls into it.  

Can see an example on reducing the speed on searched assistance - line 108

**Comment by YunataSavior** [comment]
Okay, so I resubbed for TBC prepatch just to test this...

https://clips.twitch.tv/BadCrypticHumanDxCat-DjJXzmByCK1rF_8R

(This is on Burning Crusade Classic servers)

I took an old lvl 60 mage and got a Dark Iron Tunneler to 16% health, then it flees.

**CONTRARY TO WHAT ALL YOU HAVE BEEN SAYING**, it did neither of the two things it would do on AzerothCore/ChromieCraft.
1. It flees in a random direction. NOT into the other mobs in the area.
2. You can DEFINITELY RUN FASTER THAN THE FLEEING MOB, EVEN AT 16% health!!!

So I was right this whole entire time...

**Comment by eTzmNcbkrng** [comment]
I just tested the same NPC on ChromieCraft and it behaved exactly the same as in your video.
Big OOOOOF, I wanted you to be right.

I reckon it's inconsistent.

**Comment by T1ti** [comment]
I did a lot of testing previously on classic, I wasn't able to find a consistent pattern if fleeing is random or toward allies, there seemed to be some randomness but I think when close to allies they will run to them(or at least more likely?)

And assaid above, fleeing works differently if they are running to an ally, so I expect there is two types of fleeing, fleeing to an ally, and normal fleeing

**Comment by T1ti** [comment]
@eTzmNcbkrng 
the video is private btw

**Comment by T1ti** [comment]
after further testing on classic, creatures are still slower even when fleeing toward allies

**Comment by T1ti** [comment]
https://www.youtube.com/watch?v=4mWrvfQ9Xzc
This video showcases the fleeing call for help, you can hear the murloc running to its ally and making a sound(proof that an event is happening)

They always make this sound whenever running to an ally, and then the creatures aggro in a fairly large AOE, not just the one creature he ran to

**Comment by T1ti** [comment]
https://www.youtube.com/watch?v=Cpeqj__7iIg

This video shows that even while running to an ally, the creature is still slowed (5.64 speed instead of 8.0)

**Comment by T1ti** [comment]
So TLDR how fleeing works on classic : 

- Fleeing doesn't change the creature's speed at all, it should still have the same speed as if it was not fleeing

**But currently AC is missing the low health speed reduction mechanic, which is about -2.5% reduced speed per 1% hp below 20%.** 

-If you FEAR a creature which doesn't have reduced move speed(above 20% hp) like with warlock's fear, the creature will run at full speed

-Creatures seem to run to other creatures while fleeing if they get close to them, but the original fleeing direction seems completely random(my personal analysis). When they do this, they stop at the creature and trigger a call for help, after a small delay the creatures will aggro and the fleeing creature will stop fleeing exactly at the same time the new creatures aggro.

**Comment by FrancescoBorzi** [comment]
what's the status of this?

**Comment by 55Honey** [comment]
> So TLDR how fleeing works on classic :
> 
> * Fleeing doesn't change the creature's speed at all, it should still have the same speed as if it was not fleeing
> * **But currently AC is missing the low health speed reduction mechanic, which is about -2.5% reduced speed per 1% hp below 20%.**
>   This is probably the msot important point
> * If you FEAR a creature which doesn't have reduced move speed(above 20% hp) like with warlock's fear, the creature will run at full speed.
> * **as a recap, fleeing state never affects the creature's speed, creature's speed is only affected by external factors like the low health speed reduction mechanic, or other slwos applied by players**
> * Creatures seem to run to other creatures while fleeing if they get close to them, but the original fleeing direction seems completely random(my personal analysis). When they do this, they stop at the creature and trigger a call for help, after a small delay the creatures will aggro and the fleeing creature will stop fleeing exactly at the same time the new creatures aggro.
> * basically, fleeing movement is compeltely RNG, UNTIL they get close to another creature, then they should? run to it and do a call for help.(previous point)

The Status is, we're missing someone to code this behaviour. @FrancescoBorzi

**Comment by FrancescoBorzi** [comment]
> > So TLDR how fleeing works on classic :
> > 
> > * Fleeing doesn't change the creature's speed at all, it should still have the same speed as if it was not fleeing
> > * **But currently AC is missing the low health speed reduction mechanic, which is about -2.5% reduced speed per 1% hp below 20%.**
> >   This is probably the msot important point
> > * If you FEAR a creature which doesn't have reduced move speed(above 20% hp) like with warlock's fear, the creature will run at full speed.
> > * **as a recap, fleeing state never affects the creature's speed, creature's speed is only affected by external factors like the low health speed reduction mechanic, or other slwos applied by players**
> > * Creatures seem to run to other creatures while fleeing if they get close to them, but the original fleeing direction seems completely random(my personal analysis). When they do this, they stop at the creature and trigger a call for help, after a small delay the creatures will aggro and the fleeing creature will stop fleeing exactly at the same time the new creatures aggro.
> > * basically, fleeing movement is compeltely RNG, UNTIL they get close to another creature, then they should? run to it and do a call for help.(previous point)
> 
> The Status is, we're missing someone to code this behaviour. @FrancescoBorzi

ok, added 30$ bounty on it and edited the original issue by adding the outcome of the discussion.

**Comment by UltraNix** [comment]
So the only bug here is that low-health mechanic. I have a question regarding to it: what about other slor effects on mobs below 20%? E.g. if mob have 10% hp and frostbolt debuff, it means that mob's speed is reduced twice: by frostbolt and then by low-health mechanic?

**Comment by T1ti** [comment]
> So the only bug here is that low-health mechanic. I have a question regarding to it: what about other slor effects on mobs below 20%? E.g. if mob have 10% hp and frostbolt debuff, it means that mob's speed is reduced twice: by frostbolt and then by low-health mechanic?

I guess it would work the same way as when a creature has multiple slows from players, I think only the most powerful works

**Comment by UltraNix** [comment]
Not all slow effects are exclusive to others.

**Comment by UltraNix** [comment]
Not all slow effects are exclusive to others.
Please, test it on Classic.

**Comment by T1ti** [comment]
Perhaps we should make a separate issue with what is needed for clarity ?
Like 1 issue specifically for the low health slow mechanic 

This nto be adressed asap, fleeing is liek the msot broken and anoying mechanic atm

**Comment by T1ti** [comment]
Also the title is wrong and outdated

**Comment by 55Honey** [comment]
Can you suggest a new title @T1ti ?
I think we're bound to this issue because of the Bounty and all the cross-links.

**Comment by T1ti** [comment]
The intended mechanic is 2.5% reduced speed per 1% health below 15%

**Comment by T1ti** [comment]
Precision : It's a global mechanic, it shouldn't apply only while fleeing

**Comment by T1ti** [comment]
and ti's below 20%, not 15%

**Comment by T1ti** [comment]
> ~I just tested the same NPC on ChromieCraft and it behaved exactly the same as in your video. Big OOOOOF, I wanted you to be right.~
> 
> ~I reckon it's inconsistent.~
> 
> EDIT: Spoke too soon, mobs flee at a faster pace if they are within range of friendlies.
> As evidenced by @YunataSavior's video, this was not the case in TBC.
> 
> This video was taken on AzerothCore rev. [40ed3a5](https://github.com/azerothcore/azerothcore-wotlk/commit/40ed3a575ba4ba3eff9d218d018de6a20b6e29db) https://www.youtube.com/watch?v=PCBBDFrYR-0
> 
> NPC I tested:
> 
> ```
> Dark Iron Dwarf attempts to run away in fear!
> Player selected NPC
> DB GUID: 11192, current GUID: 21995.
> Faction: 54.
> npcFlags: 0.
> Entry: 1051.
> DisplayID: 825 (Native: 825).
> Level: 27.
> EquipmentId: 1 (Original: 1).
> Health (base): 800. (max): 800. (current): 132.
> Unit Flags: 557056.
> Unit Flags 2: 2048.
> Dynamic Flags: 4.
> Faction Template: 54.
> SpawnTime: Full:5m Remain:0s
> Loot: 1051 Pickpocket: 1051 Skinning: 0
> InstanceID: 0
> Phasemask: 1
> Armor: 1078
> Position: -2844.992188 -2856.754883 33.356915.
> AIName: SmartAI ScriptName: 
> MechanicImmuneMask: 0
> ```
> 
> Here's his speed if he flees outside of his allies range:
> 
> ```
> [16:55:27] 3.5024838447571
> [16:55:27] 3.4829020500183
> [16:55:28] 0
> [16:55:28] 3.4817657470703
> [16:55:28] 3.4817657470703
> [16:55:28] 3.4817657470703
> [16:55:29] 3.4841544628143
> [16:55:29] 3.4841544628143
> [16:55:29] 3.4841544628143
> [16:55:29] 0
> [16:55:30] 0
> [16:55:30] 0
> [16:55:30] 0
> [16:55:31] 3.4818525314331
> [16:55:31] 3.4818525314331
> [16:55:31] 3.489898443222
> [16:55:31] 3.489898443222
> [16:55:32] 3.489898443222
> [16:55:32] 3.489898443222
> [16:55:32] 3.489898443222
> [16:55:32] 3.489898443222
> [16:55:32] 3.489898443222
> ```
> 
> Here's his speed if he flees within range of allies
> 
> ```
> [16:56:33] 8.0103225708008
> [16:56:33] 8.0103225708008
> [16:56:33] 8.0103225708008
> [16:56:33] 8.0103225708008
> [16:56:33] 8.0103225708008
> [16:56:33] 8.0103225708008
> [16:56:34] 8.0103225708008
> [16:56:34] 8.0103225708008
> [16:56:34] 8.0103225708008
> [16:56:34] 8.0103225708008
> ```

And if this confirmed, it's also not correct.
Creatures shouldn't have increased speed when fleeing to allies, perhaps it should be a separate issue ?
there's like 5 different issues there

**Comment by T1ti** [comment]
Ok I did further testing in AC to analsye how it currently works, as I only analysed classic before : 
in AC creatures currently don't ALWAYS flee toward allies, there is a range check it is just extremly large. from my tessting it seems to be about 40 yards. this deosn't happen in classic, they run in a  random direction like in fear and they stop and do a call for help if they get close to another creature
- I can't confirm that creatures run faster toward allies in AC, to me it seems like they go at the normal maximum speed, BUT they are much slower and walk if they don't have an ally around. This is not correct at all in classic creatures have the same speed regardless

**### TLDR of the 3 main issues :** 
- the global slow mechanic if creatures are low life isn't implemented in AC.(-2.5% speed per missing 1% hp below 20% hp). This slow penalty(and all other slow effects) should also apply if the creature is fleeing

- in AC creatures run slower if there is no ally in range(about 40 yards)  : **Not blizzlike** . The speed shouldn't change

- in AC  creatures seem to **always** flee toward allies if there is one within about 40 yards : **not blizzlike** . In classic they always run in a random direction and if they are about within  range of another creature(about 5/10 yards), they stop and do a call for help

**Comment by T1ti** [comment]
Should we open an issue for each ? currently the title is only about the first point

**Comment by ratkosrb** [comment]
There are two kinds of Fleeing. In Classic all mobs flee in random direction, while in TBC I noticed mobs running towards nearest ally. Don't know if its an universal change though.

**Comment by T1ti** [comment]
> There are two kinds of Fleeing. In Classic all mobs flee in random direction, while in TBC I noticed mobs running towards nearest ally. Don't know if its an universal change though.

I was doing my testing on the TBC PTR with low lvl mobs

**Comment by YunataSavior** [comment]
> 
> 
> There are two kinds of Fleeing. In Classic all mobs flee in random direction, while in TBC I noticed mobs running towards nearest ally. Don't know if its an universal change though.

By "TBC", are you referring to "outlands", or "TBC Classic"?

**Comment by YunataSavior** [comment]
https://youtu.be/HsnZpGBaGKo?t=90

The above video was taken from retail TBC classic in Outlands.

Look VERY carefully at the top-right. You'll see a friendly that's WELL WITHIN range of the peons that the player is fighting (in terms of the 40 yard check). A few seconds into combat, one of the peons runs away in fear. It does NOT run into its friendly!

I think by default, let's make it such that, for every mob type, they do not beeline their friendlies, UNLESS we can produce **SOLID REAL EVIDENCE** that the original behavior is indeed intentional on the real servers (on a case-by-case basis).

**Comment by T1ti** [comment]
I tested it a lot and unless the creature is extremly close, like within 10 yards it seems to always run in a random direction

**Comment by patou01** [comment]
@T1ti This is not fixed as far as I understand.
Are we confident in "2.5% per 1% missing hp"?
In the video you linked:
the creature is still slowed (5.64 speed instead of 8.0)
5.64/8 = 0.7, so it was slowed 30% for 7% hp under 20%. ie 1% = 30/7 = 4.2% per hp? I recall mobs becoming extremely slow at a few percents, but at the same time, I think they lose speed more progressively than a cut-off at 20%.

I have a somewhat working piece of code, but need the following info:
- does this apply to all mobs, or are there known exceptions (bosses? I think they used to slow down as well)
- Is the formula of 2.5% per % below 20% correct? My understanding is that this formula would give a speed reduction of 17.5% (2.5*7, as he has 13% hp) on the murloc you showed, however we are seeing almost double that.
- when the same mob is healed, should he turn back to his normal speed? (my memories from official wow is yes)

Then I will need to evaluate what happens when slows are stacked, to confirm I didn't make a hack.

**Comment by patou01** [comment]
Found some unclear code. I expect this is partly trying to do what we refer to in this issue. I will be checking a bit around to see if I can change it to work.
![image](https://user-images.githubusercontent.com/2592673/133135415-d384fc9f-17de-4d0e-9f10-0d32fcf41076.png)

Should this slow from hp stack with slow from auras? (eg frostbolt)

**Comment by T1ti** [comment]
> Found some unclear code. I expect this is partly trying to do what we refer to in this issue. I will be checking a bit around to see if I can change it to work.
> ![image](https://user-images.githubusercontent.com/2592673/133135415-d384fc9f-17de-4d0e-9f10-0d32fcf41076.png)
> 
> Should this slow from hp stack with slow from auras? (eg frostbolt)

I already reported that AC is currently applying a speed reduction when the creature is earching for assistance 
![image](https://user-images.githubusercontent.com/40864460/133151112-0b445942-e9ca-4443-b322-26b36f0677e9.png)

**Comment by T1ti** [comment]
> @T1ti This is not fixed as far as I understand.
> Are we confident in "2.5% per 1% missing hp"?
> In the video you linked:
> the creature is still slowed (5.64 speed instead of 8.0)
> 5.64/8 = 0.7, so it was slowed 30% for 7% hp under 20%. ie 1% = 30/7 = 4.2% per hp? I recall mobs becoming extremely slow at a few percents, but at the same time, I think they lose speed more progressively than a cut-off at 20%.
> 
> I have a somewhat working piece of code, but need the following info:
> 
> * does this apply to all mobs, or are there known exceptions (bosses? I think they used to slow down as well)
> * Is the formula of 2.5% per % below 20% correct? My understanding is that this formula would give a speed reduction of 17.5% (2.5*7, as he has 13% hp) on the murloc you showed, however we are seeing almost double that.
> * when the same mob is healed, should he turn back to his normal speed? (my memories from official wow is yes)
> 
> Then I will need to evaluate what happens when slows are stacked, to confirm I didn't make a hack.

Update, I did additional testing on TBC classic and it is below 30%, not 20%...

The reduction seems to be 0.0166...(2.5/1.5) per level

![image](https://user-images.githubusercontent.com/40864460/133159979-a0c12215-bbb8-4456-a80d-0572fbaeb211.png)

My formula seems to match pretty well the values measured on classic. Note that speed varies quite a lot on classic(by up to 5%ish), based on if they are turning, using animations etc so it is hard to get perfect values.

**Comment by 55Honey** [comment]
Thanks T1ti, i'm updaring the title based on your findings.

**Comment by patou01** [comment]
So I have a way of doing this, however it might increase the server load a decent amount (scaling linearly with the amount of mobs engaged on the server), so that could be a concern. Also maybe this slow should be an aura in terms of code, but that's something I don't know.

I am not sure if this slow should stack however, mobs hit by a frostbolt at 5% hp seem to become **extremely** slow, if you have the possibility, do you mind checking that?

**Comment by T1ti** [comment]
perhaps trigger the slow on creature hp changed

btw I suspect that on classic it's not exactly 1.66% per lvl, imo it is progressive in digits and they do the exact math calculation, like if at 14.54% hp :` 8 - ( 30 - 14.54) * ...`

**Comment by T1ti** [comment]
![image](https://user-images.githubusercontent.com/40864460/133314128-dedd9b14-b161-49ee-ac62-c06ba32c0c08.png)
the slow does stack, here's a frostbolt

**Comment by T1ti** [comment]
the value looks very close to your video, your stacking formula is probably correct
(3.2 at 9% with frostbolt)

**Comment by patou01** [comment]
@T1ti, I believe this issue is fully fixed isn't it?
